Trading v3.0.0

andiahlm opened the pull request.

andiahlm committed:
add-first-draft-new-entity-model

andiahlm committed:
Refactor execution relations

TLusti committed:
fix missing array typing for lists

TLusti committed:
fix wrong reference executionAllocation.transactionId

TLusti committed:
fix make order.status readOnly

andiahlm committed:
Update bundled tradingAPI.yaml

TLusti committed:
fix missing array typing for list in accountaccess endpoint

andiahlm committed:
Various updates

andiahlm committed:
Add order status endpoint and fix account endpoint

andiahlm committed:
Fix descriptions

andiahlm committed:
Execution endpoints and various refactorings

andiahlm committed:
fix orderId

andiahlm committed:
Add base vs quote instrument to order

andiahlm committed:
Fix typos

andiahlm committed:
refactor and split api

andiahlm committed:
add support for contract poposals and quotes and fx samples

andiahlm committed:
fixes on descriptions

andiahlm committed:
Refactor quotes and financial instruments

andiahlm committed:
Add expiry to quote

andiahlm committed:
fix contract quote

andiahlm committed:
Fix financial instrument

andiahlm committed:
various backlog v3

andiahlm committed:
Fixes after rebase

andiahlm committed:
Add precious metal

TLusti commented on src/trading/components/schemas/Account.yaml:
Should include an attribute to indicate wheter there is only read access or read-write access. Could potentially also be defined as an array, otherwise not sufficient for [https://github.com/OpenWealth/OpenWealth/issues/118](url)

andiahlm replied:
accessType property added

TLusti commented on src/trading/components/schemas/Order.yaml:
```diff
@@ -0,0 +1,124 @@
+      This is the time in seconds of a 'goodForTime' TimeInForce.
+      Positive integer value which represents the time is seconds in which the new order
+      remains active in the market before it is automatically cancelled (e.g. expired).
+    type: integer
```
example missing

TLusti commented on src/trading/components/schemas/Order.yaml:
```diff
@@ -0,0 +1,124 @@
+    type: integer
+  quoteId:
+    type: string
+    description: Unambiguous identifier for the quote that was provided by the confirming party.
```
example missing

andiahlm replied:
added example and maxLength

TLusti commented on src/trading/components/schemas/QuoteBase.yaml:
```diff
@@ -0,0 +1,40 @@
+      Kind of quote requested.
+      - Price quote TBD
+      - Contract quote TBD
+    example: buy
```
example not matching enum

TLusti commented on src/trading/components/schemas/ContractQuote.yaml:
```diff
@@ -0,0 +1,9 @@
+title: Standard quote
```
Title and name of scheme not consistent, could lead to confusion

TLusti commented on src/trading/components/schemas/CurrencySwap.yaml:
```diff
@@ -0,0 +1,10 @@
+title: Forward Currency
```
Wrong title and description, this is a Swap not a Fwd

TLusti commented on src/trading/components/schemas/CurrencySwap.yaml:
```diff
@@ -0,0 +1,10 @@
+title: Forward Currency
+description: Forward Currency Instrument
```
Wrong title and description, this is a Swap not a Fwd

TLusti commented on src/trading/components/schemas/ContractQuote.yaml:
```diff
@@ -0,0 +1,9 @@
+  - $ref: QuoteBase.yaml
+  - type: object
+    properties:
+      quotedContract:
```
Should be read-only in my opinion

andiahlm replied:
agree

TLusti commented on src/trading/components/schemas/Allocation.yaml:
clientReference missing on allocation, would be needed to suffice [https://github.com/OpenWealth/OpenWealth/issues/38](url)

TLusti replied:
CancelledQuantity missing on allocation level
[https://github.com/OpenWealth/OpenWealth/issues/170](url)

TLusti replied:
executedQuantity is missing on allocation

andiahlm replied:
First point will not be implemented as agreed

TLusti commented on src/trading/components/schemas/Order.yaml:
Order Placer and Decision Maker are missing as attributes as per [https://github.com/OpenWealth/OpenWealth/issues/47](url)

andiahlm replied:
Do we have any definition of the properties given by finma?

andiahlm replied:
Definition to be added later

TLusti commented on src/trading/Docs/Enums.md:
```diff
@@ -97,7 +97,6 @@
 | cancelledByOther | Any other party than instructing party or hub cancelled the order (e.g. exchange, broker, etc.) |
 | cannotContainMarketIdentificationAndMarketDescription | Place of trade cannot contain market identification and market description at the same time |
```
adjusted enum name not reflected

andiahlm replied:
Documentation to be updated when moved to wiki

TLusti commented on src/trading/Docs/Enums.md:
```diff
@@ -122,14 +121,13 @@
-| orderQuantityTypeNotImplemented | The selected type of the orderQuantity is not implemented |
+| orderQuantityTypeNotSupported | The selected type of the orderQuantity is not supported |
 | orderQuantityTypeWrong | The selected type of the orderQuantity does not match the selected financialInstrument |
 | orderTypeNotSupportedByMarket | Given order type is not valid for given market |
```
adjusted enum name not reflected

andiahlm replied:
Documentation to be updated when moved to wiki

TLusti commented on src/trading/components/schemas/NotificationOrderPlacement.yaml:
Notification was not adjusted according to newest release on STFI [https://github.com/swissfintechinnovations/ca-shared/tree/main/webhooks](url) as per issue [https://github.com/OpenWealth/OpenWealth/issues/175](url)

TLusti commented on src/trading/components/parameters/executionId.yaml:
```diff
@@ -0,0 +1,7 @@
+schema:
+  type: string
+  maxLength: 20
+description: The executionId as given py the API provider.
```
typo in description

TLusti commented on src/trading/components/parameters/expandOrder.yaml:
never used in spec

andiahlm replied:
param added to orders endpoints

TLusti commented on src/trading/components/parameters/orderId.yaml:
```diff
@@ -0,0 +1,7 @@
+schema:
+  type: string
+  maxLength: 20
+description: The orderId as given py the API provider.
```
typo in description

TLusti commented on src/trading/components/schemas/AccountAccess.yaml:
never used in spec

TLusti commented on src/trading/components/schemas/AccruedInterest.yaml:
```diff
@@ -6,10 +6,9 @@ properties:
   amount:
-    $ref: 'amount.yaml'
+    type: number
+    description: TBD.
```
Proper description missing

TLusti replied:
e.g. The monetary amount of interest that has been earned or accumulated on an account, investment, or financial instrument up to a specific point in time, but has not yet been paid or received.

andiahlm replied:
Added

TLusti commented on src/trading/components/schemas/Allocation.yaml:
```diff
@@ -0,0 +1,51 @@
+  quantity:
+    description: The requested quantity to be allocated to specific safekeeping account.
+    $ref: 'Quantity.yaml'
+  remainingQuantity:
```
was removed on order level, to be consistent, should be removed here as well. But only in combination with the introduction of cancelledQuantity.

TLusti commented on src/trading/components/schemas/BillingDetails.yaml:
according to the meeting on 20.02.2025, (for the non-fee or tax attributes) only 

- grossTradeAmount: Principal amount of a trade (price multiplied by quantity).
- settlementAmount: Total amount of money to be paid or received in exchange for the securities. The amount includes the principal with any commissions and fees or accrued interest.

should be covered

TLusti commented on src/trading/components/schemas/NotificationId.yaml:
Notification was not adjusted according to newest release on STFI [https://github.com/swissfintechinnovations/ca-shared/tree/main/webhooks](https://github.com/OpenWealth/OpenWealth/pull/198/url) as per issue [https://github.com/OpenWealth/OpenWealth/issues/175](https://github.com/OpenWealth/OpenWealth/pull/198/url)

TLusti commented on src/trading/components/schemas/EventSubscription.yaml:
Notification was not adjusted according to newest release on STFI [https://github.com/swissfintechinnovations/ca-shared/tree/main/webhooks](https://github.com/OpenWealth/OpenWealth/pull/198/url) as per issue [https://github.com/OpenWealth/OpenWealth/issues/175](https://github.com/OpenWealth/OpenWealth/pull/198/url)

TLusti commented on src/trading/components/schemas/EventSubscriptionId.yaml:
Notification was not adjusted according to newest release on STFI [https://github.com/swissfintechinnovations/ca-shared/tree/main/webhooks](https://github.com/OpenWealth/OpenWealth/pull/198/url) as per issue [https://github.com/OpenWealth/OpenWealth/issues/175](https://github.com/OpenWealth/OpenWealth/pull/198/url)

TLusti commented on src/trading/components/schemas/EventSubscriptionRequest.yaml:
Notification was not adjusted according to newest release on STFI [https://github.com/swissfintechinnovations/ca-shared/tree/main/webhooks](https://github.com/OpenWealth/OpenWealth/pull/198/url) as per issue [https://github.com/OpenWealth/OpenWealth/issues/175](https://github.com/OpenWealth/OpenWealth/pull/198/url)

TLusti commented on src/trading/components/schemas/EventType.yaml:
Notification was not adjusted according to newest release on STFI [https://github.com/swissfintechinnovations/ca-shared/tree/main/webhooks](https://github.com/OpenWealth/OpenWealth/pull/198/url) as per issue [https://github.com/OpenWealth/OpenWealth/issues/175](https://github.com/OpenWealth/OpenWealth/pull/198/url)

TLusti commented on src/trading/components/schemas/ExchangeTradedDerivative.yaml:
not all attributes outlined specified in: https://github.com/OpenWealth/OpenWealth/issues/70

TLusti commented on src/trading/components/schemas/ExecutionAllocation.yaml:
```diff
@@ -0,0 +1,27 @@
+type: object
+description: Execution allocation
+properties:
+  # allocationId:
```
Why was this "commented out"?

andiahlm replied:
We do not yet consider the execution allocation a resource and therefore not model relations via id

TLusti commented on src/trading/components/schemas/FinancialInstrumentIdentification.yaml:
```diff
@@ -0,0 +1,31 @@
+      - sicovam
+      - belgian
+      - common
+      - cryptoCurrency
```
better naming: digitalAssetTicker

TLusti commented on src/trading/components/schemas/FxForwardContract.yaml:
```diff
@@ -0,0 +1,20 @@
+allOf:
+  - $ref: 'FinancialInstrumentBase.yaml'
+  - type: object
+    required:
```
attribute names in required not consistent with actual property names

TLusti commented on src/trading/components/schemas/PositionEffect.yaml:
```diff
@@ -0,0 +1,9 @@
+type: string
+description: |
+  Indicates whether the resulting position after a trade should be an opening position or closing position. 
+  sed for omnibus accounting - where accounts are held on a gross basis instead of being netted together.
```
typo? Otherwise, I do not understand the second part of the description

andiahlm replied:
Copy paste issue from Omix, removed second part

TLusti commented on src/trading/paths/event-subscriptions_{eventSubscriptionId}.yaml:
Notification was not adjusted according to newest release on STFI [https://github.com/swissfintechinnovations/ca-shared/tree/main/webhooks](https://github.com/OpenWealth/OpenWealth/pull/198/url) as per issue [https://github.com/OpenWealth/OpenWealth/issues/175](https://github.com/OpenWealth/OpenWealth/pull/198/url)

TLusti commented on src/trading/paths/event-subscriptions_{eventSubscriptionId}_event-notifications.yaml:
Notification was not adjusted according to newest release on STFI [https://github.com/swissfintechinnovations/ca-shared/tree/main/webhooks](https://github.com/OpenWealth/OpenWealth/pull/198/url) as per issue [https://github.com/OpenWealth/OpenWealth/issues/175](https://github.com/OpenWealth/OpenWealth/pull/198/url)

TLusti commented on src/trading/paths/orders_{orderId}.yaml:
```diff
@@ -0,0 +1,38 @@
+  summary: Returns a single order
+  description: Returns a specific security order.
+  operationId: getOrderById
+  parameters:
```
expand missing?

andiahlm committed:
various fixes

andiahlm committed:
Add access type property to account

andiahlm committed:
add expandOrdert param to path

andiahlm committed:
fix settlement amount on billing

andiahlm committed:
Remove netBillingAmount

andiahlm committed:
Add algo trade instructions

andiahlm committed:
Fix required on algo trade instruction

andiahlm committed:
Add algo trade schemas

andiahlm committed:
Update event notifications

andiahlm committed:
Fix tag naming convention

andiahlm committed:
fix tag

andiahlm committed:
Add flat algo trade

andiahlm committed:
Add single, swap and mulileg orders & quotes, customer

andiahlm committed:
Bundled api

andiahlm committed:
multi-leg refactoring

- multi-leg related refactoring
- various fixes descriptions, conventions

andiahlm committed:
Various multileg, enum and cancellation related

andiahlm committed:
Fix on quantity description

andiahlm committed:
Add order leg index

TLusti commented on src/trading/components/examples/order.json:
Example not in line anymore with the current spec

andiahlm replied:
Removed the examples for now from the spec. Examples will be updated accordingly

TLusti commented on src/trading/paths/orders.yaml:
```diff
@@ -0,0 +1,87 @@
+  description: Posts a new order.
+  operationId: createOrder
+  parameters:
+    - $ref: ../components/parameters/expandOrder.yaml
```
Expand should not be on POST but on the GET /orders

TLusti commented on src/trading/components/schemas/Account.yaml:
```diff
@@ -0,0 +1,43 @@
+type: object
+required:
+  - type
+  - iban
```
IBAN cannot be mandatory, as Safekeeping accounts dont have ibans usualy

TLusti commented on src/trading/components/schemas/Account.yaml:
```diff
@@ -0,0 +1,43 @@
+type: object
+required:
```
I would expect id to be required, as this is the key they should use for requests the specific resource

TLusti commented on src/trading/API.yaml:
```diff
@@ -1,484 +1,61 @@
+    $ref: paths/customers.yaml
+  /customers/{customerId}:
+    $ref: paths/customers_{customerId}.yaml
+  /accounts:
```
Best practice would be to also add a /accounts/{accountId} endpoint to be consistent

TLusti commented on src/trading/components/schemas/EventNotification.yaml:
Not latest version of notifications & subscriptions

andiahlm replied:
Added the latest from our PR to ca-shared

TLusti commented on src/trading/components/schemas/EventSubscription.yaml:
Not latest version of notifications & subscriptions

TLusti commented on src/trading/components/schemas/EventType.yaml:
Not latest version of notifications & subscriptions

andiahlm replied:
This ok:
```
description: Event type the subscription should apply to AND reason why a notification is sent.
type: string
enum:
  - heartbeat
  - order:doneForDay
  - order:statusUpdated
example: order:statusUpdated
```

TLusti commented on src/trading/components/schemas/Order.yaml:
```diff
@@ -0,0 +1,141 @@
+    description: Optional reference to link orders, i.e. multi-leg orders.
+    readOnly: true
+  bestEffortExecution:
+    description: TBD
```
Description missing. 
Proposal: If this flag is set to true, then the broker proceeds with the placement of the bulk order in case if one or more clients identified within allocations list of the given order by corresponding account id fail pre-trade checks and are thus excluded from the order. The failed client allocations are excluded from the order before the actual moment of placement, and total quantity of the order is adjusted accordingly to exclude quantities request for given client allocations.

In case if the parameter is set to false (or equally, not explicitly passed), then the order is rejected by the broker if at least one client fails pre-trade checks

TLusti commented on src/trading/components/schemas/TraderInstruction.yaml:
```diff
@@ -0,0 +1,88 @@
+      Sets the minimum fill size in dark venues client may want.
+    type: number
+  primaryOnly:
+    description: TBD.
```
Description Missing.
Proposal: Indicates that the order should be solely traded in primary market.

TLusti commented on src/trading/components/schemas/TraderInstruction.yaml:
```diff
@@ -0,0 +1,88 @@
+      - neutral
+      - aggressive
+      - superAggressive
+  minQty:
```
Would adjust wording to minQuantity to stay consistent in the naming, as we did not abbreviate Quantity in any other place

TLusti commented on src/trading/components/parameters/expandOrder.yaml:
```diff
@@ -0,0 +1,19 @@
+  items:
+    type: string
+    enum:
+      - allocationList
```
If allocationList is either way mandatory, an expand on this in the GET /orders endpoints only make limited sense.

andiahlm replied:
Decided to make it optional to support expansion

TLusti commented on src/trading/components/schemas/Execution.yaml:
```diff
@@ -0,0 +1,31 @@
+type: object
+description: Entity describing a market execution
```
Required Fields missing, would expect at least the orderId to be required

andiahlm replied:
These introduced as required:
```
required:
  - id
  - orderId
  - reference
  - quantity
```

TLusti commented on src/trading/components/schemas/Order.yaml:
```diff
@@ -0,0 +1,141 @@
+type: object
+description: Order object.
+required:
```
legList needs to be required

andiahlm committed:
Fix on quote and order type

andiahlm committed:
Various restructure

- account to accountId on allocation
- replacement of quantity and price schemas
- error enums and type enums

TLusti commented on src/trading/components/parameters/orderId.yaml:
```diff
@@ -0,0 +1,7 @@
+required: true
+schema:
+  type: string
+  maxLength: 20
```
inconsistent with orderId - Length should be 128

TLusti commented on src/trading/components/schemas/Account.yaml:
```diff
@@ -0,0 +1,46 @@
+  - type
+  - number
+  - customerId
+properties:
```
currency property as optional missing

TLusti commented on src/trading/components/schemas/CancellationReason.yaml:
```diff
@@ -0,0 +1,65 @@
+            - other
+        propertyPath:
+          type: string
+          description: Path to affected property according to TBD
```
Description still incl. TBD - would go for JSONPath as a recommendation

TLusti commented on src/trading/components/schemas/Customer.yaml:
```diff
@@ -0,0 +1,19 @@
+title: Customer
+type: object
+description: Overview of the customer with the respective accounts.
```
No "respective accounts" in this scheme anymore

TLusti commented on src/trading/components/schemas/EventNotification.yaml:
```diff
@@ -0,0 +1,22 @@
+  version:
+    type: string
+    example: '1.0'
+  eventType:
```
would reference to EventType.yaml

TLusti commented on src/trading/components/schemas/EventNotification.yaml:
```diff
@@ -0,0 +1,22 @@
+    example: 7c5f40fc-1b29-4263-a2dd-e127a22a947f
+  resourceLink:
+    type: string
+    example: /api/bankingservices/b-link/order-placement/v1/orders/abcd1234
```
no bLink references in OW API

TLusti replied:
still bLink in example

TLusti commented on src/trading/components/schemas/EventNotification.yaml:
```diff
@@ -0,0 +1,22 @@
+    type: string
+    example: /api/bankingservices/b-link/order-placement/v1/orders/abcd1234
+  createdAt:
+    type: string
```
format: date-time should be added

TLusti commented on src/trading/components/schemas/EventSubscription.yaml:
```diff
@@ -0,0 +1,20 @@
+  callbackUrl:
+    type: string
+    example: https://ttp.com/API/notifications/order-placement/
+    description: Not to be used in bLink
```
change description: e.g. The callback URL is the destination endpoint where our system will send real-time event notifications. Provide a publicly accessible HTTPS URL that can accept POST requests with JSON payloads.

TLusti commented on src/trading/components/schemas/EventNotification.yaml:
```diff
@@ -0,0 +1,22 @@
+    maxLength: 36
+    example: 7c5f40fc-1b29-4263-a2dd-e127a22a947f
+  resourceLink:
+    type: string
```
description missing:

e.g. 

The resourceLink is a direct URL reference to the specific resource or object that triggered the webhook event. This link provides immediate access to the full details of the affected item in the source system.

TLusti commented on src/trading/components/schemas/ExecutionAllocation.yaml:
```diff
@@ -0,0 +1,29 @@
+  executionIds:
+    description: List of market executions, that contributed to the allocated quantity.
+    items:
+      type: string
```
would it make sense to define the executionId separately and reference it, as we did in all other occasions?

andiahlm replied:
I tried to remove all primitives, especially if they are used only once

TLusti commented on src/trading/components/schemas/FinancialInstrument.yaml:
```diff
@@ -0,0 +1,52 @@
+type: object
+required:
+  - type
```
no "type" property on instrument anymore - therefore should be removed from required field

TLusti commented on src/trading/components/schemas/Order.yaml:
```diff
@@ -0,0 +1,132 @@
+type: object
+description: Order object.
+required:
```
would expect id to be both readOnly & required. This should be allowed after: [https://swagger.io/docs/specification/v3_0/data-models/data-types/](url)

_If a readOnly or writeOnly property is included in the required list, required affects just the relevant scope – responses only or requests only. That is, read-only required properties apply to responses only, and write-only required properties – to requests only._

andiahlm replied:
I read somewhere that it is not a good practice since many code generators don't handle it well...

TLusti commented on src/trading/components/schemas/Order.yaml:
```diff
@@ -0,0 +1,132 @@
+  groupReference:
+    type: string
+    maxLength: 128
+    description: Optional reference to link orders, i.e. multi-leg orders.
```
would adjust the description to: Optional reference to link orders, i.e. strategies

TLusti commented on src/trading/components/schemas/Order.yaml:
```diff
@@ -0,0 +1,132 @@
+    readOnly: true
+  status:
+    $ref: OrderStatus.yaml
+  cancellationReasonList:
```
After discussions, this should be renamed to something closer to cancellationOrRejectionReasonList

TLusti commented on src/trading/components/schemas/CancellationReason.yaml:
```diff
@@ -0,0 +1,65 @@
+type: object
```
After discussions, this should be renamed to something closer to cancellationOrRejectionReason

TLusti commented on src/trading/components/schemas/OrderType.yaml:
Should be removed as per discussions

TLusti commented on src/trading/components/schemas/Quote.yaml:
```diff
@@ -0,0 +1,34 @@
+  - type
+  - legList
+properties:
+  id:
```
would expect it to be readOnly and required

andiahlm replied:
see above

TLusti commented on src/trading/components/schemas/Quote.yaml:
```diff
@@ -0,0 +1,34 @@
+    description: List of unique and unambiguous identification (i.e. UUID) used by the bank for the customer.
+    type: array
+    items:
+      type: string
```
would refer to items as separate objects that are referenced, instead of hard typed in here

andiahlm replied:
We leave it as is, as we currently try not to create schemas for primitives

TLusti commented on src/trading/components/schemas/TraderInstruction.yaml:
```diff
@@ -0,0 +1,88 @@
+      - liquidityGrabber
+      - darkOnly
+      - custom
+  # startTime:
```
either remove # or use duration similar to timeInForce. Preference would be duration

TLusti replied:
after discussion: introduce all three attributes: startTime, stopTime and Duration for maximum flexibility

TLusti commented on src/trading/paths/event-subscriptions.yaml:
```diff
@@ -0,0 +1,93 @@
+          schema:
+            type: object
+            properties:
+              eventNotifications:
```
should be eventSubscriptions not eventNotifications, but title not necessary at all in my opinion

TLusti commented on src/trading/paths/event-subscriptions_{eventSubscriptionId}.yaml:
```diff
@@ -0,0 +1,105 @@
+  description: Get event subscription by id.
+  parameters:
+    - $ref: ../../generics/components/parameters/x_correlation_id_in_header.yaml
+  operationId: getEventSubscription
```
is this operationsId should rather be getEventSubscriptionById

TLusti commented on src/trading/paths/event-subscriptions_{eventSubscriptionId}_event-notifications.yaml:
```diff
@@ -0,0 +1,48 @@
+get:
+  summary: Get event notifications list
+  description: Get aggregated event-notification list.
+  operationId: getEventNotifications
```
Proposal:
operationId: getEventNotificationsByEventSubscriptionId

TLusti commented on src/trading/paths/event-subscriptions_{eventSubscriptionId}_event-notifications.yaml:
```diff
@@ -0,0 +1,48 @@
+              eventNotifications:
+                type: array
+                items:
+                  $ref: ../components/schemas/EventSubscription.yaml
```
wrong reference, should point to EventNotification.yaml

TLusti commented on src/trading/paths/orders.yaml:
```diff
@@ -0,0 +1,86 @@
+      application/json:
+        schema:
+          $ref: ../components/schemas/Order.yaml
+        # examples:
```
remove for final release if no examples are present

TLusti commented on src/trading/paths/orders.yaml:
```diff
@@ -0,0 +1,86 @@
+    - order
+  summary: Posts a new order
+  description: Posts a new order.
+  operationId: createOrder
```
why not postOrder to stay consistent with previous version?

andiahlm replied:
many use CRUD names for operation ids since put is not so intuitive

TLusti commented on src/trading/paths/orders_{orderId}.yaml:
```diff
@@ -0,0 +1,39 @@
+        application/json:
+          schema:
+            $ref: ../components/schemas/Order.yaml
+          # examples:
```
remove examples if not provided

TLusti commented on src/trading/paths/orders_{orderId}_executions.yaml:
```diff
@@ -0,0 +1,42 @@
+  summary: Returns a list of executions by order
+  description: |
+    Returns all executions by order ID
+  operationId: getOrderEcecutions
```
typo in operationId - 
Would rather call it getExecutionsByOrderId

TLusti commented on src/trading/components/schemas/AccruedInterest.yaml:
```diff
@@ -6,10 +6,9 @@ properties:
     example: 28
```
type: number ; format: integer 
is not a valid combination. Either its type: number or type: integer

type: integer can have three formats:

type | format | Description
-- | -- | --
number | – | Any numbers.
number | float | Floating-point numbers.
number | double | Floating-point numbers with double precision.
integer | – | Integer numbers.
integer | int32 | Signed 32-bit integers (commonly used integer type).
integer | int64 | Signed 64-bit integers (long type).

Source: https://swagger.io/docs/specification/v3_0/data-models/data-types/#numbers

andiahlm committed:
Various linting and descriptions

TLusti commented on src/trading/components/parameters/expand_order.yaml:
```diff
@@ -0,0 +1,19 @@
+  - `allocationList`: Include allocationList details.
+  - `allocationList.executionAllocationList`: Include executionAllocationList details on the allocation.
+  - `allocationList.allocationCancellationReasonList`: Include allocationCancellationReasonList details on the allocation.
+  - `orderCancellationReasonList`: Include orderCancellationReasonList
```
naming inconsistent with new object name

TLusti commented on src/trading/components/schemas/CancellationOrRejectionReason.yaml:
```diff
@@ -0,0 +1,69 @@
+            - other
+        propertyPath:
+          type: string
+          description: Path to affected property according to TBD
```
Description still missing

TLusti commented on src/trading/components/schemas/ExcerciseStyle.yaml:
```diff
@@ -0,0 +1,8 @@
+type: string
+description: Specifies how the derivatice instrument can be exercised.
```
typo in derivative

TLusti commented on src/trading/components/schemas/UnitType.yaml:
```diff
@@ -0,0 +1,6 @@
+description: Unit type of the measure of the instrument (i.e. prcious metal).
```
typo in description : i.e. precious metal

TLusti commented on src/trading/API.yaml:
```diff
@@ -1,484 +1,91 @@
+    $ref: paths/orders.yaml
+  /orders/{orderId}:
+    $ref: paths/orders_{orderId}.yaml
+  /orders/{orderId}/action/cancel:
```
Would it make sense to call it "actions" instead of "action" ?

TLusti commented on src/trading/API.yaml:
```diff
@@ -1,484 +1,91 @@
-  version: 2.2.1
-  title: Order Placement API
-  description: This API is used to place orders and check on the order status.
+  version: 3.0.0-draft
```
set to 3.0.0 with release candidate

andiahlm committed:
various

andiahlm committed:
fix operationIds

andiahlm committed:
Add operation id guide

andiahlm committed:
various linting

andiahlm committed:
Fix cosmetics

TLusti reviewed (approved):
LGTM - only comment is to adjust version before merging

andiahlm committed:
fix samples and cleanup

andiahlm committed:
fix casing on error response

andiahlm committed:
fix casing correlation id

actions-user committed:
Update bundled API file

actions-user committed:
Update bundled API file

andiahlm committed:
Update webhook api

andiahlm committed:
Adapt sfti header and parameter naming convention

andiahlm committed:
Notification and header related fixes

andiahlm committed:
Fixes on headers

andiahlm committed:
Fixes quotes and descriptions

andiahlm committed:
Add bundled

actions-user committed:
Update bundled API file

TLusti reviewed (approved):
lgtm

andiahlm committed:
Fix param naming conventions

TLusti committed:
Update OperationIdGuide.md

Fixed typos

andiahlm committed:
Add valueDate to execution allocation

andiahlm committed:
Add samples

- customers
- accounts
- quote
- order
- execution

andiahlm committed:
Add fx swap samples

- add fx swap samples
- fix bond samples

andiahlm committed:
Various fixes

andiahlm committed:
fix path variables

andiahlm committed:
Fix tag and bundle

andiahlm committed:
Various small fixes and bundle

andiahlm committed:
bundle webhook api

andiahlm committed:
Fiy types and format
